bin/jsondiff.py

```diff
@@ -0,0 +1,113 @@
+                         type=validate_json_extension,
+                         nargs=2)
+    # Note: nargs='?' and const=value is for 0 or 1 arguments supplied. If 0 arguments provided, the const value will be used
+    options.add_argument('--rel_tol',
+                         metavar='<relative_error>',
+                         dest='relative_error',
+                         help='Specify the maximum relative error (default 1e-8)',
+                         type=float,
+                         nargs='?', 
+                         const=1e-8)
+    options.add_argument('--abs_tol',
+                         metavar='<absolute_error>',
+                         dest='absolute_error',
+                         help='Specify the maximum absolute error (default 1e-10)',
+                         type=float,
+                         nargs='?', 
+                         const=1e-10)
+    options.add_argument('-h', '--help', action='help', help='Displays CLI usage statement')
+    args, unknown = parser.parse_known_args()
+    
+    # Set arg parameter to None if no flag was supplied
+    if '--rel_tol' not in sys.argv[1:]:
+        args.relative_error = None
+    if '--abs_tol' not in sys.argv[1:]:
+        args.absolute_error = None
+    return args
+
+def validate_json_extension(json_file):
+    """Validates that the json file has an .json extension and path exists
+        json_file: the file to validate"""
+    return_code = 0
+    base, extension = os.path.splitext(json_file)
+    if extension.lower() not in ('.json'):
+        raise argparse.ArgumentTypeError('Invalid extension with "' + str(json_file) + '": Provide a file with a .json extension')
+    return json_file
+
+def validate_json_files_exist(args):
+    """Check if json files exist in the system path
+        args: arguments provided by the user"""
+    return_code = 0
+    are_files = list()
+    for json_file in args.json_files:
+        if not os.path.exists(json_file):
+            print('Path does not exist: ' + str(json_file))
+            return_code += 1
+    return return_code
+
+def validate_flags(args):
+    """Validate that only one flag was passed via the command-line interface
+        args: arguments provided by the user"""
+    return_code = 0
+    if args.relative_error is None and args.absolute_error is None:
+        return_code += 1
+        print('Error: No flag provided. Use "-h" or "--help" for more information.')
+    elif args.relative_error is not None and args.absolute_error is not None:
+        return_code += 1
+        print('Error: One flag expected but two flags were provided. Use "-h" or "--help" for more information.')
+    return return_code
+ 
+def main():
+    return_code = 0
+    
+    # Create command-line arguments
+    args = add_arguments()
+    
+    # Validation of command-line arguments
+    return_code += validate_json_files_exist(args)
+    return_code += validate_flags(args)
```

Comment:
These should call the functions above and raise an exception, so you will not need to check the return codes from these. The exception will trigger a non-zero exit code.